src/mqtt/qmqtt_client_p.cpp

```diff
@@ -274,6 +276,15 @@ void QMQTT::ClientPrivate::onTimerPingReq()
 {
     Frame frame(PINGREQ);
     _network->sendFrame(frame);
+    _pingResponseTimer.start();
+}
+
+void QMQTT::ClientPrivate::onPingTimeout()
+{
+    Q_Q(Client);
+    _pingResponseTimer.stop();
+    disconnectFromHost();
+    emit q->pingTimeout();
```

Comment:
IMO, this should fire a connection error prior disconnectFromHost()

I don't see any benefits of having pingresp() and pingTimeout() signals

Reply:
Yes, this would simplify the patch, I agree.

Reply:
The pingresp signal was already there, so I do not want to remove it. Instead of the pingTimeout signal I propose introducing a new error state and emit the error signal. That will allow the user of the class to find out why he was disconnected.